### PR TITLE
IOW-527 RowHandler and Tab Delimiter update

### PR DESCRIPTION
**Pull request by duselmann**
Before making a pull request
----------------------------

- [x] Make sure all tests run
- [x] Update the changelog appropriately

Description
-----------
Contains code review recommendations to use jdbctemplate row handler and improve the separation char logic. The row handler is not expected to return anything and is the appropriate interface for writing the file. The sepChar logic is replaced with a list of string joined with tab.

After making a pull request
---------------------------
- [x] If appropriate, put the link to the PR in the JIRA ticket
- [x] Assign someone to review unless the change is trivial

### CODE REVIEW

**Review by ssoper-usgs — APPROVED** [review]

File: src/main/java/gov/usgs/wma/waterdata/groundwater/RdbWriter.java [hunk 1]
```diff
@@ -65,49 +66,51 @@ public RdbWriter writeHeader() {
...
+		writeRow(columns);
+		dataLineCount++;
+		return this;
+	}
+
+	protected RdbWriter writeRow(List<String> columns) {
 		try {
-			rdb.append("\n");
+			String row = String.join(DELIMITER, columns.toArray(new String[] {}));
+			rdb.append(row).append("\n");
```
Comment (ee-usgs) [inline src/main/java/gov/usgs/wma/waterdata/groundwater/RdbWriter.java:110]: Nice improvement!

File: src/main/java/gov/usgs/wma/waterdata/groundwater/RdbWriter.java [hunk 2]
```diff
@@ -118,20 +121,15 @@ public RdbWriter writeRow(DiscreteGroundWater dgw) {
...
+		// Trim the value to the proper field length.
+		String trimmedValue = value;
+		if (value == null) {
+			value = "";
 		}
-		return this;
+		if (value.length() > length) {
+			trimmedValue = value.substring(0, length);
+		}
+		return trimmedValue;
```
Comment (ee-usgs) [inline src/main/java/gov/usgs/wma/waterdata/groundwater/RdbWriter.java:133]: Another nice addition

File: src/main/java/gov/usgs/wma/waterdata/groundwater/DiscreteGroundWaterRowMapper.java [hunk 3]
```diff
@@ -40,8 +35,6 @@ public String mapRow(ResultSet rs, int rowNum) throws SQLException {
 		discreteGroundWater.timezoneCode = rs.getString("timezone_code");
 		discreteGroundWater.verticalDatumCode = rs.getString("vertical_datum_code");
 		discreteGroundWater.parameterCode = rs.getString("parameter_code");
-
-		writer.writeRow(discreteGroundWater);
-		return "written";
+		return discreteGroundWater;
```
Comment (ee-usgs) [inline src/main/java/gov/usgs/wma/waterdata/groundwater/DiscreteGroundWaterRowMapper.java:38]: I think this is much clearer in intent.

**Review by ee-usgs — APPROVED** [review]
Nice improvements overall - thanks Dave!

**Pull request merged**